v-next/hardhat/src/internal/core/config-validation.ts

```diff
@@ -457,6 +472,7 @@ export function validatePositionalArguments(
 
     if (arg.defaultValue !== undefined) {
       switch (arg.type) {
+        case ArgumentType.STRING_WITHOUT_DEFAULT:
         case ArgumentType.STRING:
         case ArgumentType.FILE: {
           if (
```

Comment:
Why do we need array checks here?
When we look at usages, arrays do not appear expressible in the type system.
The message suggests you can set to an array of strings but the type system blocks it.